submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,244 @@
+      k8s.ovn.org/ovn-zone: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.1/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster2-worker
+```
+
+* Multiple Zone: In a multiple-zone setup, we will have an OVN database and an OVN master pod for each zone. Transit switches connect the
+  zones. The OVN-Kubernetes services ensure that the necessary routes are added for pod and service reachability across nodes in different
+  zones.
+
+```bash
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.3/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.5/16"]'
+      k8s.ovn.org/ovn-zone: az2
+    name: cluster2-worker
+```
+
+With the current architecture, Submariner adds routes only in the zone in which it is deployed. For example, if Submariner is deployed in
+zone 1 it programs OVN db in zone 1. So only pods in zone 1 nodes will be able to talk to other clusters. Pods in zone 2 or zone 3 will not
+be able to reach remote clusters connected via Submariner.
+
+As part of the proposal, we plan to support both the modes and OVN cluster deployments where interconnect
+is not enabled as well.
+
+### Route APIs
+
+As part of this proposal, we are planning to add two new CRDs in Submariner. The plan is to decouple OVN dataplane programming so that this
+can be moved to OVN, when OVN provides a way to add custom routes. We can propose this CRD or consume the CRD that they provide at a later
+point of time.
+
+#### SubmarinerGwRoute
+
+This CR will be created when a remote endpoint is added and there will be one CR per remote cluster. This CRD has two fields
+
+* NextHops - Specifies the list of next hop to reach the remote cluster, in this case it will be the IP of ovn-k8s-mp0
+  interface, the interface used by OVN for host networking.
+* RemoteCIDR - Specifies the list of  remote CIDRs reachable via this cluster.
+
+This CR will be used by the route agent pod running on the active-Gateway node to program OVN to send the traffic destined to
+remote clusters via the Submariner tunnel.
+
+``` go
+type SubmarinerGwRoute struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    SubmarinerRoutePolicySpec SubmarinerRoutePolicySpec `json:"submarinerRoutePolicySpec"`
+}
+
+type SubmarinerRoutePolicySpec struct {
+    //Specifies the list of next hops to reach the remote CIDRs
+    NextHops []string `json:"nextHops"`
+
+    //Specifies the remote CIDRs available via the next hop
+    RemoteCidr []string `json:"remoteCidr"`
+}
+```
+
+#### SubmarinerNonGWRoute
+
+This CR will be created when a remote endpoint is created and there will be one created per endpoint.
+
+* NextHops - Specifies the list of next hops. In this case it will be the transit switch IP.
```

Comment:
Please add a note saying that in the proposal only a single nextHop will be used and it will point to the transit switch ip of the zone in which the active Gateway is present.

Reply:
Done